core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -1210,8 +1221,15 @@ public ScheduledFuture<?> scheduleInEventLoop(Runnable runnable, long duration,
         }, duration, timeUnit);
     }
 
+    public void updateTickingState(float tickRate, boolean frozen) {
+        tickThread.cancel(false);
+        this.tickingFrozen = frozen;
+        millisecondsPerTick = Math.round(1000.0f / MathUtils.clamp(tickRate, 1.0f, 10000.0f));
```

Comment:
is the maximum arbitrary? didn't see one in TickRateManager

the only reason i ask is because if we are going set a max on it, we may as well ensure that the msPerTick isn't gonna be rounded to zero

Reply:
The maximum is what the server says the maximum is when you do /tick rate 99999999

If the vanilla client allows for higher tick rates then ill just make it a minimum cap only

Reply:
Alright cool. We should still make sure that the rounding cannot produce 0 though—can't imagine the event loop likes being provided a period of zero. Could set a minimum on the millisecondsPerTick, or use nanosecondsPerTick instead.